containers/docker/container_update.go

```diff
@@ -166,8 +166,29 @@ func (m *Manager) ContainerUpdate(ctx context.Context, instance, image, tag, cmd
 	// All checks passed, proceed to the actual (synchronous or asynchronous) update.
 	if async {
 		klog.Infof("Starting asynchronous update of instance %s to image %s:%s with cmd %s and options %+v", instance, image, tag, cmd, opts)
-		// There can only be one go routine per instance name due to the mutex handling.
-		go m.performContainerUpdate(ctx, instance, image, tag, cmd, cnts, opts...)
+		deadline, ok := ctx.Deadline()
+		// Override the cancellation from the parent context.
+		// This allows the RPC to exit while the async update completes.
+		ctx = context.WithoutCancel(ctx)
```

Comment:
Thank you for this - I wonder though - should we have a default deadline in case the parent context does not have one? It seems this would hang if the update does not complete.

Reply:
sure, yep its good to prevent the server from getting blocked here.
Although as this would be a default i think the default should be something relatively excessive like 5 minutes just to be on the safe side.
If the client wanted a shorter/longer timeout they could set it themselves, but 5 minutes should hopefully be enough to stop/start a container (especially as no image pull is needed)

Reply:
Yup that sounds good to me - thank you for this!

Reply:
added a default deadline of 5 minutes in https://github.com/openconfig/containerz/pull/30/commits/6b429c26048e13002132c47515e81ff57a6fad6c